### ISSUE TITLE
Box selection stops working when transitioning to (x2, y2, z2) 

### ISSUE THREAD

**Issue by serpens24** [constellation-app/constellation#389]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help` > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/wiki/FAQ
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

You can manually transition node positions between (x, y, z) and (x2, y2, z2) using the Page-Up and Page-Down keys. However, doing so results in box selection no longer working if the nodes are at any position other than (x, y, z). Click selection works regardless of the node positions. 
(Editors node - does page up/down, toggle between (x, y, z) and (x2, y2, z2) .. or zoom in/out graph, I suspect its a form of zoom in / out)

### Steps to Reproduce

1. Crerate a new empty graph
2. Use Edit->Draw Mode and add several nodes to the graph at positions othewr than 0 (say [10,10], [10, -10], [20, 5])
3. Press page up 5 or 6 times
4. Use rectangle select to make a tight selection around the nodes and Note that not all nodes (if any) are selected
5. Scroll the mouse wheel down so that entire graph is zoom way out and nodes are mere specks
6.Use rectangle seelction to make a large selection of all of graph and note the nodes are selected

From the above steps, it appears the coordinates the rectangle selection is extracting from the graph, and the coordinates the nodes are using are differnet meaning interesction tests are failing

**Expected behaviour:**
Step 4 should select all nodes

**Actual behaviour:** 
Step 4 selected 0 nodes

**Reproduces how often:**
100%

### Additional Information

NIL

**Comment by serpens24** [comment]
I'm quite suspicious of the following code in BoxSelectionPlugin.java

```
            // If mixing is required then mix the main location with the alternative location.
            if (requiresMix) {
                x = inverseMix * x + mix * graph.getFloatValue(x2Attr, vxId);
                y = inverseMix * y + mix * graph.getFloatValue(y2Attr, vxId);
                z = inverseMix * z + mix * graph.getFloatValue(z2Attr, vxId);
            }

            <snip>

            if (sceneLocation.getZ() < 0) {
                final float horizontalOffset = sceneLocation.getX() / -sceneLocation.getZ();
                final float verticalOffset = sceneLocation.getY() / -sceneLocation.getZ();

                if (horizontalOffset > left && horizontalOffset < right && verticalOffset < top && verticalOffset > bottom) {
                    vxIncluded.set(vxId);
                }
            }
```
I found that this is firing often and I found the value of inverseMix to have a value of 9 in my example, now because x2/y2/z2 attributes were all set to 0, the calcualtions above are pretty much just multiplying x/y/z by inverseMix. So a value of x or -2 transaltes to -18 (for instance).

Now, if the above multiplication factors occur, then the tests comparing horizontalOffset (a factor of x) to left will fail as in my tests the values of left, right, top, bottom are the same irrespective of the zoom in and out I do with PgUp/PgDwn.

So in essence we are comparing offsets deived from vertex positions with left/right/top/bottom positions, the offsets get adjusted hen zooming but the left/right/top/bottom values do not ... this mismatch is the basis of the error.

I dont have an understanding of the graph transformations taking place in this code ATM, bit the incompatibilities described above are clear.

**Comment by serpens24** [comment]
Note also from above, the value of mix  increased by 1 every time I press the PgUp button ..... (starting at 0).

**Comment by antares1470** [comment]
I think I've worked out what the issue is here. The main problem here is the mix variable in the execute function of BoxSelectionPlugin.java

When I did some testing, the mix variable ranges in value from 0-20. On the other hand, the code seems to assume that the mix variable ranges in value from 0-1 (hence the reason I believe mix is a float not an int). This can also be seen in the following code:
`if (requiresMix && mix == 0.0f) {
            requiresMix = false;
 } else if (requiresMix && mix == 1.0f) {
            xAttr = x2Attr;
            yAttr = y2Attr;
            zAttr = z2Attr;
            requiresMix = false;
 }`
What this code suggests is that when the mix is 1, a vertex is at the (x2, y2, z2) coordinates when in reality this doesn't happen until mix is 20. Having experimented changing the else if condition to `requiresMix && mix == 20.0f` I can box select properly when the mix is 20.

I don't know if the code has changed elsewhere to cause this problem but it would seem to adjusting the code for the new mix range should fix the problem
